### PR TITLE
Add shapefile import

### PR DESCRIPTION
**Pull request by basilveerman**

### CODE REVIEW

**Comment by basilveerman** [comment]
Does not currently perform error/complexity checking. Added issue #25 to address this.

File: src/core/geo.js [hunk 1]
```diff
@@ -104,14 +96,41 @@ var g = {
...
+
+  load: function (file, success) {
+    /* All load functions must call `success` handler with a GeoJSON feature */
+    var ext = file.name.split('.')[1];
+
+    if (_.contains(['geojson', 'json'], ext)) {
+      this.loadTextFormat(file, success);
+    } else if (ext === 'zip') {
+      this.loadShapefile(file, success);
+    }
```
Comment (jameshiebert) [inline src/core/geo.js]:
```js
else {
// inform the user about what they should do?
}
```

File: src/core/geo.js [hunk 2]
```diff
@@ -104,14 +96,41 @@ var g = {
...
+      shp(evt.target.result).then(function (geojson) {
+        success(geojson);
+      });
+    };
+
+    reader.readAsArrayBuffer(file);
+  },
+
+  load: function (file, success) {
+    /* All load functions must call `success` handler with a GeoJSON feature */
```
Comment (jameshiebert) [inline src/core/geo.js]: It's a *little* confusing reading the code with the success handler being passed down two levels. As long as that stack doesn't get any deeper (i.e. as long as the `load[type]file` methods don't do any more asynchronous operations) then we're probably OK. If it does get any deeper with another pass through, we may want to reorganize a bit. I think it's good for now, though.

**Comment by jameshiebert** [comment]
Once you add an `else` clause in the `load()` function, I think that it's good to go.

**Pull request merged**